jax/_src/dtypes.py

```diff
@@ -685,3 +685,13 @@ def check_user_dtype_supported(dtype, fun_name=None):
     fun_name = f"requested in {fun_name}" if fun_name else ""
     truncated_dtype = canonicalize_dtype(dtype).name
     warnings.warn(msg.format(dtype, fun_name , truncated_dtype), stacklevel=3)
+
+def check_ndarray_for_int_overflow(x):
```

Comment:
I'd annotate this `x : np.ndarray` just to make clear that this is meant for numpy array inputs.

Reply:
Great idea.